src/main/java/com/twilio/oai/ConventionMap.java

```diff
@@ -0,0 +1,100 @@
+package com.twilio.oai;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+
+public class ConventionMap {
+    final static String SEGMENT_HYDRATE = "hydrate";
+    final static String SEGMENT_SERIALIZE = "serialize";
+    final static String SEGMENT_DESERIALIZE = "deserialize";
+    final static String SEGMENT_LIBRARY = "library";
+    final static String SEGMENT_TYPES = "types";
+    final Map<String, Map<String, Object>> conventionalMapping = new HashMap<>();
+
+    public ConventionMap() {
+        conventionalMapping.put(SEGMENT_HYDRATE, new HashMap<>());
+        conventionalMapping.put(SEGMENT_SERIALIZE, new HashMap<>());
+        conventionalMapping.put(SEGMENT_DESERIALIZE, new HashMap<>());
+        conventionalMapping.put(SEGMENT_LIBRARY, new HashMap<>());
+        conventionalMapping.put(SEGMENT_TYPES, new HashMap<>());
+        init();
+    }
+
+    public void add(String segment, String key, Object value) {
+        conventionalMapping.get(segment).put(key, value);
+    }
+
+    public Map<String, Object> get(String segment) {
+        return conventionalMapping.get(segment);
+    }
+
+    public Object get(String segment, String key) {
+        return conventionalMapping.get(segment).get(key);
+    }
+
+    private void init() {
+        initHydrate();
+        initSerialize();
+        initDeserialize();
+        initLibrary();
+        initTypes();
+    }
+
+    private void initHydrate() {
+        this.add(SEGMENT_HYDRATE,"date", "DateConverter.localDateFromString(%s)");
+        this.add(SEGMENT_HYDRATE,"date-time-rfc-2822", "DateConverter.rfc2822DateTimeFromString(%s)");
+        this.add(SEGMENT_HYDRATE,"date-time", "DateConverter.iso8601DateTimeFromString(%s)");
+    }
+
+    private void initSerialize() {
+        this.add(SEGMENT_SERIALIZE,"endpoint","{%s}.getEndpoint()");
+        this.add(SEGMENT_SERIALIZE,"iso-country-code","%s");
+        this.add(SEGMENT_SERIALIZE,"object","Converter.mapToJson(%s)");
+        this.add(SEGMENT_SERIALIZE,"date","DateConverter.dateStringFromLocalDate(%s)");
+        this.add(SEGMENT_SERIALIZE,"date-time","%s.toInstant().toString()");
+        this.add(SEGMENT_SERIALIZE,"date-or-time","%s.toInstant().toString()");
+        this.add(SEGMENT_SERIALIZE,"string","%s");
+        this.add(SEGMENT_SERIALIZE,"sid","%s");
+    }
+
+    private void initDeserialize() {
+        this.add(SEGMENT_DESERIALIZE,"currency", "com.twilio.converter.CurrencyDeserializer");
+    }
+
+    private void initLibrary() {
+        this.add(SEGMENT_LIBRARY, "currency", Arrays.asList("java.util.Currency"));
```

Comment:
May consider moving these mappings into a config file, similar to how we map types in api-defs.

Reply:
> I dislike these changes continuing to go to a non-main branch. Need to prioritize getting to a state where we can merge to main.

We created story for that and we will deprecate this branch by this iteration.